.github/workflows/build-docker.yml

```diff
@@ -79,6 +81,48 @@ jobs:
           platform=${{ matrix.platform }}
           echo "PLATFORM_TUPLE=${platform//\//-}" >> $GITHUB_ENV
 
+      - name: Docker apt & pip caches
+        uses: actions/cache@v4
+        id: docker-caches
+        with:
+          path: |
+            var-cache-apt
+            var-lib-apt
+            root-cache-pip
+          key: docker-caches-${{ matrix.platform }}-${{ hashFiles('Dockerfile') }}
+
+      - name: Docker Cargo caches
+        uses: actions/cache@v4
+        id: docker-cargo-caches
+        with:
+          path: |
+            root-target
+            usr-local-cargo-git-db
+            usr-local-cargo-registry
+          key: docker-cargo-caches-${{ matrix.platform }}-${{ hashFiles('Dockerfile', 'crates/**', 'Cargo.toml', 'Cargo.lock') }}
```

Comment:
You _could_ drop the crates directory from the hash key and so reuse the cargo cache even when source code files change. This would give you incremental builds.

Reply:
I'm not sure we _want_ incremental builds in the release pipeline. A clean build seems like a good property, right?

Reply:
That's why I picked this config, but I just wanted to point out the option.